### PR TITLE
Add 3D (NCW) channels-last dim_order support to Neutron runtime (#19341)

### PR DESCRIPTION
**Pull request by JakeStevens**
Summary:

Although 3dim dim order is not supported pytorch, it actually *is* properly supported within the NXP backend's implementation.

The only issue is within the runtime, which forces the check to four dimensions.

This PR relaxes and supports 3D.

Differential Revision: D102862305

### CODE REVIEW

**Review by jirioc — APPROVED** [review]
The implementation looks good. Note that sometimes the transpositions are done by the neutron instead of the backend (scheduled by AoT), so I you need to make sure that that part is also compatible with 3D.

**Review by MartinPavella — APPROVED** [review]

**Comment by MartinPavella** [comment]
LGTM. But perhaps some test with a simple 3D channels last dim order model would be useful.

**Comment by JakeStevens** [comment]
> LGTM. But perhaps some test with a simple 3D channels last dim order model would be useful.

Are there examples in the repo of cpp tests that use the backend cpp file?

**Comment by MartinPavella** [comment]
> Are there examples in the repo of cpp tests that use the backend cpp file?

You can use a test like this:
```python
class Conv1DModel(torch.nn.Module):

    def __init__(self):
        super().__init__()
        self.conv = torch.nn.Conv1d(2, 2, 3)

    def forward(self, x):
        return self.conv(x)


def test_3d_channels_last_dim_order__conv():
    model = Conv1DModel().eval().to(memory_format=torch.channels_last)

    # Use the `channels last` dim order.
    input_spec = [
        ModelInputSpec((1, 2, 5), dim_order=torch.channels_last)
    ]

    comparator = NumericalStatsOutputComparator(max_mse_error=8e-3)
    lower_run_compare(
        model,
        input_spec,
        dataset_creator=RandomDatasetCreator(),
        output_comparator=comparator,
        dlg_model_verifier=BaseGraphVerifier(1, []),
        reference_model=ReferenceModel.QUANTIZED_EDGE_PYTHON
    )
```

File: backends/nxp/tests/generic_tests/test_3d_channels_last.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        self.conv = torch.nn.Conv1d(2, 2, 3)
+
+    def forward(self, x):
+        return self.conv(x)
+
+
+def test_3d_channels_last_dim_order__conv(mocker):
+    model = Conv1DModel().eval()
+
+    input_spec = [ModelInputSpec((1, 2, 5), dim_order=torch.channels_last)]
```
Comment (MartinPavella) [inline backends/nxp/tests/generic_tests/test_3d_channels_last.py:33]: According to your modifications to `executorch_pipeline`, the example input used to export this model will use the `contiguous` memory format. Also, you are not calling `model.to(memory_format=torch.channels_last) here. If I understand correctly, this is because `torch` doesn't support 3D channels last memory format. But how will the `dim_order` entry in the nodes' `meta` attribute be set? I.e. how will ExecuTorch know to use the 3D channels last dim_order?

To me, it looks like the test is just using regular `contiguous` dim_order, and the specified `torch.channels_last` in the input spec is being silently ignored.

Is my assumption correct, or am I missing something?
